opta/core/plan_displayer.py

```diff
@@ -0,0 +1,116 @@
+    ),
+    SERIOUS_SEVERITY: (
+        "This is the highest level of severity and this means that either you are destroying something or made a big "
+        "change which forced something to recreate (destroy and create itself). Please tread carefully and know what "
+        "you are doing."
+    ),
+}
+
+
+def _max_severity(severity_1: str, severity_2: str) -> str:
+    if SERIOUS_SEVERITY in [severity_1, severity_2]:
+        return SERIOUS_SEVERITY
+    if MODERATE_SEVERITY in [severity_1, severity_2]:
+        return MODERATE_SEVERITY
+    return BENIGN_SEVERITY
+
+
+class PlanDisplayer:
+    @staticmethod
+    def display(detailed_plan: bool = False) -> None:
+        if detailed_plan:
+            Terraform.show(TF_PLAN_PATH)
+            return
+        plan_dict = Terraform.show_plan()
+        plan_severity = BENIGN_SEVERITY
+        module_changes: dict = {}
+        resource_change: dict
+        for resource_change in plan_dict.get("resource_changes", []):
+            if resource_change.get("change", {}).get("actions", ["no-op"]) == ["no-op"]:
+                continue
+            address: str = resource_change["address"]
+
+            if not address.startswith("module."):
+                logger.warn(
+                    f"Unable to determine severity of changes to resource {address}. "
+                    "Please run in detailed plan mode for more info"
+                )
+            module_name = address.split(".")[1]
+            module_changes[module_name] = module_changes.get(
+                module_name, {"severity": BENIGN_SEVERITY, "resources": {}}
+            )
+            resource_name = ".".join(address.split(".")[2:])
+            actions = resource_change.get("change", {}).get("actions", [])
+            if "create" in actions and "delete" in actions:
+                actions = ["replace"]
+            action = actions[0]
+            current_severity = ACTION_TO_SEVERITY[action]
+            action_reason = resource_change.get("action_reason", "N/A")
+            module_changes[module_name]["resources"][resource_name] = {
+                "action": action,
+                "reason": action_reason,
+                "severity": current_severity,
+            }
+            module_changes[module_name]["severity"] = _max_severity(
+                module_changes[module_name]["severity"], current_severity
+            )
+            plan_severity = _max_severity(plan_severity, current_severity)
+
+        logger.info(
+            f"Identified total severity of {SEVERITY_COLORS[plan_severity]}{plan_severity}{attr(0)}.\n"
+            f"{SEVERITY_EXPLANATIONS[plan_severity]}\n"
+            "If you want extra help, please feel free to reach out to the Runx team at https://slack.opta.dev.\n"
+            "Severity break down by module is as follows:"
+        )
+        for module_name, module_change in module_changes.items():
+            current_severity = module_change["severity"]
+            logger.info(
```

Comment:
Couple of ideas/improvements:
1) Sort the module_changes.items based on severity (SERIOUS,MODERATE,BENIGN)
2) You may want to present it in a tabular format by working the the logger.info string - e.g. like so https://www.educba.com/python-print-table/

Reply:
I agree with the sorting and will implement it now, but I'm not sure that the tabular format looks nicer.

Reply:
sounds good